src/mam/mod_mam_pm.erl

```diff
@@ -260,23 +255,58 @@ filter_packet({From, To, Acc1, Packet}) ->
                                          from_jid => From,
                                          to_jid => To }, Acc3),
     Acc5 = mod_amp:check_packet(Acc4, AmpEvent),
-    {From, To, Acc5, mongoose_acc:element(Acc5)}.
+    {ok, {From, To, Acc5, mongoose_acc:element(Acc5)}}.
 
 process_incoming_packet(From, To, Packet, Acc) ->
     handle_package(incoming, true, To, From, From, Packet, Acc).
 
 %% hook handler
--spec remove_user(mongoose_acc:t(), jid:user(), jid:server()) -> mongoose_acc:t().
-remove_user(Acc, User, Server) ->
+-spec remove_user(Acc, Params, Extra) -> {ok, Acc} when
+      Acc :: mongoose_acc:t(),
+      Params :: #{jid := jid:jid()},
+      Extra :: map().
+remove_user(Acc, #{jid := #jid{luser = User, lserver = Server}}, _) ->
     delete_archive(Server, User),
```

Comment:
The call function here could also be changed, `delete_archive/2` will later do `jid:make_bare/2` and normalise the strings, when they're already normalised. If it's not too complex, it could already be improved here, and have the functions down the callstack use the already normalised `#jid{}` record

Reply:
done 👍